tools/rebuildEM/main.cpp

```diff
@@ -77,6 +78,22 @@ int main(int argc, char** argv)
         }
     }
 
+    // Check that `mariadb-columstore` is in inactive state.
+    const std::string checkColumnstoreService =
+        "if systemctl status mariadb-columnstore | grep -q "
+        "\"inactive\"\n then\n exit 1\n else\n exit 0\n fi\n";
```

Comment:
There's no need to do this grep and shell. Look at man systemctl and the exit status is:

```
       ┌──────┬──────────────────────────────────────────┬─────────────────────────────────────┐
       │Value │ Description in LSB                       │ Use in systemd                      │
       ├──────┼──────────────────────────────────────────┼─────────────────────────────────────┤
       │0     │ "program is running or service is OK"    │ unit is active                      │
       ├──────┼──────────────────────────────────────────┼─────────────────────────────────────┤
       │1     │ "program is dead and /var/run pid file   │ unit not failed (used by is-failed) │
       │      │ exists"                                  │                                     │
       ├──────┼──────────────────────────────────────────┼─────────────────────────────────────┤
       │2     │ "program is dead and /var/lock lock file │ unused                              │
       │      │ exists"                                  │                                     │
       ├──────┼──────────────────────────────────────────┼─────────────────────────────────────┤
       │3     │ "program is not running"                 │ unit is not active                  │
       ├──────┼──────────────────────────────────────────┼─────────────────────────────────────┤
       │4     │ "program or service status is unknown"   │ no such unit                        │
       └──────┴──────────────────────────────────────────┴─────────────────────────────────────┘
```

So you can use that exit status directly.

As a better invocation without output and minimal examination of unneeded things:`systemctl is-active --quiet mariadb-columnstore.service`

Reply:
Thx for the note @grooverdan

Reply:
@grooverdan thanks!